fenicsxprecice/fenicsxprecice.py

```diff
@@ -172,18 +173,21 @@ def write_data(self, write_function):
                 CouplingMode.BI_DIRECTIONAL_COUPLING)
 
         w_func = write_function
+        write_function_type = determine_function_type(write_function)
 
         # Check that the function provided lives on the same function space provided during initialization
-        assert (self._write_function_type == determine_function_type(w_func))
-        # TODO this raises AssertionError, not sure why. I just commented it out, still works...
-        # assert (write_function.function_space == self._write_function_space)
+        assert (self._write_function_type == write_function_type)
+        # TODO: fails for weird reasons.
+        #assert (write_function.function_space == self._write_function_space)
```

Comment:
Lets figure this out before we merge this, it looks like we are messing up some types in DOLFINx